site/src/pages/tutorials/grpc/blog/implement-read.mdx

```diff
@@ -0,0 +1,280 @@
+  ```java filename=BlogClient.java
+  class BlogClient {
+    static BlogServiceBlockingStub client;  // added before
+    static BlogServiceBlockingStub cb;      // add this one here
+  }
+  ```
+2. Create a client instance with <type://Clients#builder>.
+  ```java filename=BlogClient.java
+  class BlogClient {
+    public static void main(String[] args) throws Exception {
+      // added before
+      client = Clients.newClient(
+                "gproto+http://127.0.0.1:8080/",
+                BlogServiceBlockingStub.class);
+      // add this one here
+      cb = Clients.builder("gproto+http://127.0.0.1:8080/")
+                  .build(BlogServiceBlockingStub.class);
+    }
+  }
+  ```
+3. Add a method each for retrieving posts. Note, this time we're using `cb`, the client instance built with <type://Clients.builder>.
+
+  <Tabs>
+  <TabPane tab="Single post" key="1">
+
+  Add a method in the BlogClient.java to retrieve a single post.
+
+  ```java filename=BlogClient.java
+  import example.armeria.server.blog.grpc.Blog.GetBlogPostRequest;
+  ...
+  static void getBlogPost(Integer id){
+    final BlogPost blogPost = cb.getBlogPost(GetBlogPostRequest.newBuilder().setId(id).build());
+  }
+  ```
+
+  </TabPane>
+  <TabPane tab="Multiple posts" key="2">
+
+  Add a method in the BlogClient.java to retrieve a list of posts.
+
+  ```java filename=BlogClient.java
+  import example.armeria.server.blog.grpc.Blog.ListBlogPostsRequest;
+  import example.armeria.server.blog.grpc.Blog.ListBlogPostsResponse;
+
+  ...
+  static void listBlogPosts() {
+    final ListBlogPostsResponse
+            response = cb.listBlogPosts(ListBlogPostsRequest.newBuilder().setOrderBy("asc").build());
+
+    final List<BlogPost> blogs = response.getBlogsList();
+  }
+  ```
+
+  </TabPane>
+  </Tabs>
+
+## 3. Test
+
+Let's start off with retrieving all the blog posts there are:
+
+1. Add the `listBlogPosts()` method in the test call.
+  ```java filename=BlogClient.java
+  static void testRun(){
+    createBlogPost("Another blog post", "Creating a post via createBlogPost().");
+    listBlogPosts();
+  }
+  ```
+2. [Re-run the server](/tutorials/grpc/blog/run-service#3-serve-the-service). Your server is running if you see the following message.
+  ```bash
+  [armeria-boss-http-*:8080] INFO com.linecorp.armeria.server.Server - Serving HTTP at /[0:0:0:0:0:0:0:0]:8080 - http://127.0.0.1:8080/
+  ```
+3. Run the client.
+
+Since we haven't added any log messages, we can't tell if we've successfully retrieved blog posts or not. To check the result, we'll add in an interceptor as the next step. Not only can we see the blog posts retrieved, we can obtain IDs to retrieve a post later on.
+
+## 4. Add interceptor
+
+We want to retrieve a single blog post, but we don't have the post ID information. (Although we very much know how the IDs are generated in this tutorial.) Here, what we'll do is intercept the response from the server to get the ID.
```

Comment:
How about telling users to use `GrpcServiceBuilder.intercept()` if they want to use gRPC interceptors?